content/en/blog/2023/native-sidecars/index.md

```diff
@@ -0,0 +1,207 @@
+These images are built from the latest Istio commit, so again - do not try in prod!
+When we install, we will enable the feature flag for native sidecar support and turn on access logs to help demo things later.
+
+{{< text shell >}}
+$ TAG=$(curl https://storage.googleapis.com/istio-build/dev/latest)
+$ curl -L https://storage.googleapis.com/istio-build/dev/$TAG/istioctl-$TAG-linux-amd64.tar.gz | tar xz
+$ ./istioctl install --set values.pilot.env.ENABLE_NATIVE_SIDECARS=true -y --set meshConfig.accessLogFile=/dev/stdout
+{{< /text >}}
+
+And finally we can deploy a workload:
+
+{{< text shell >}}
+$ kubectl label namespace default istio-injection=enabled
+$ kubectl apply -f samples/sleep/sleep.yaml
+{{< /text >}}
+
+Let's look at the pod:
+
+{{< text shell >}}
+$ kubectl get pods
+NAME                     READY   STATUS    RESTARTS   AGE
+sleep-7656cf8794-8fhdk   2/2     Running   0          51s
+{{< /text >}}
+
+Everything looks normal at first glance...
+If we look under the hood, we can see the magic, though.
+
+{{< text shell >}}
+$ kubectl get pod -o "custom-columns="\
+"NAME:.metadata.name,"\
+"INIT:.spec.initContainers[*].name,"\
+"CONTAINERS:.spec.containers[*].name"
+
+NAME                     INIT                     CONTAINERS
+sleep-7656cf8794-8fhdk   istio-init,istio-proxy   sleep
+{{< /text >}}
+
+Here we can see all the `containers` and `initContainers` in the pod.
+
+Surprise! `istio-proxy` is now an `initContainer`.
+
+More specifically, it is an `initContainer` with `restartPolicy: Always` set (a new field, enabled by the `SidecarContainers` feature).
+This tells Kubernetes to treat it as a sidecar.
+
+This means that later containers in the list of `initContainers`, and all normal `containers` will not start until proxy container is first ready.
+Additionally, the pod will terminate even if the proxy container is still running.
+
+### Init container traffic
+
+To put this to the test, let's make our pod actually do something.
+Here we deploy a simple pod that sends a request in an `initContainer`.
+Normally, this would fail.
+
+{{< text yaml >}}
+apiVersion: v1
+kind: Pod
+metadata:
+  name: sleep
+spec:
+  initContainers:
+  - name: check-traffic
+    image: istio/base
+    command:
+    - curl
+    - httpbin.org/get
+  containers:
+  - name: sleep
+    image: istio/base
+    command: ["/bin/sleep", "infinity"]
+{{< /text >}}
+
+Checking the proxy container, we can see the request both succeeded and went through the Istio sidecar:
+
+{{< text shell >}}
+$ kubectl logs sleep -c istio-proxy | tail -n1
+[2023-07-25T22:00:45.703Z] "GET /get HTTP/1.1" 200 - via_upstream - "-" 0 1193 334 334 "-" "curl/7.81.0" "1854226d-41ec-445c-b542-9e43861b5331" "httpbin.org" ...
+{{< /text >}}
+
+### Exiting pods
+
+Earlier, we mentioned that when applications exit (common in `Jobs`), the Pod would live forever.
+Fortunately, this is addressed as well!
+
+First we deploy a pod that will exit after one second and doesn't restart:
+
```

Comment:
Suggest using the deploy command and highlight this is deployed to the default ns that has injection enabled.